lib/baseClasses/HttpError.js

```diff
@@ -22,21 +22,26 @@ function HttpError() {
 
     // normalize the signatures
     var parsed = helpers.parseVariadicArgs(arguments, true);
-    var options = parsed.options;
     var args = parsed.args;
+    var options = parsed.options || {};
 
-    // inherit from WError, call super first before doing anything else!
+    // inherit from WError, call super first before doing anything else!  WError
+    // will handle calling captureStackTrace, using arguments.callee to
+    // eliminate unnecessary stack frames.
     WError.apply(self, args);
 
     /**
      * the error message.
      * when message isn't passed in with options, expect WError constructor
      * to fill in the message for us with the passed through args,
-     * usually via sprintf or just straight strings.
+     * usually via sprintf or just straight strings. thus, only set it
```

Comment:
I'll leave a comment making this clearer, but essentially it boils down to the variadic args. User can opt to pass in `options.message`, but ALSO follow that up with printf style messages that pass through to VError. It's basically this signature:

new Error(options [, sprintf, args...])

In that scenario, VError will generate the printf message for us, and we should prefer that over the `options.message` value. I made a note about it in README, but let me document the code too.